### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by lemosdeveloper**
Taks done:

- Update fabpot/goutte to accept version 4.0+
- Test with phpunit
- updated README

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -32,7 +34,7 @@
 
     "autoload-dev": {
         "psr-4": {
-            "Behat\\Mink\\Tests\\Driver\\": "tests"
+            "Behat\\Mink\\Driver\\Tests\\": "tests"
```
Comment (aik099) [inline composer.json:37]: Why you've changed the namespace?

File: composer.json [hunk 2]
```diff
@@ -17,11 +17,13 @@
...
         "php":                           ">=5.4",
         "behat/mink-browserkit-driver":  "~1.2@dev",
-        "fabpot/goutte":                 "~1.0.4|~2.0|~3.1"
+        "fabpot/goutte":                 "~1.0.4|~2.0|~3.1|^4.0"
     },
 
     "require-dev": {
-        "mink/driver-testsuite": "dev-master"
+        "mink/driver-testsuite": "dev-master",
+        "phpunit/phpunit": "^9.0",
```
Comment (aik099) [inline composer.json:25]: Please remove, because PHPUnit is already indirectly required via `mink/driver-testsuite`. Locking to PHPUnit 9.x would also break builds for PHP versions not supported by the PHPUnit 9.x.

File: composer.json [hunk 3]
```diff
@@ -17,11 +17,13 @@
...
+        "guzzlehttp/guzzle": "^7.0"
```
Comment (aik099) [inline composer.json:26]: Why this was added?

File: phpunit.xml.dist [hunk 4]
```diff
@@ -2,18 +2,21 @@
...
     <testsuites>
-        <testsuite name="Driver test suite">
+        <!-- <testsuite name="Driver test suite">
             <directory>tests</directory>
             <directory>vendor/mink/driver-testsuite/tests/Basic</directory>
             <directory>vendor/mink/driver-testsuite/tests/Form</directory>
+        </testsuite> -->
+        <testsuite name="TesteSuite">
+            <directory>./tests</directory>
         </testsuite>
```
Comment (aik099) [inline phpunit.xml.dist:12]: Why this change was made?

File: phpunit.xml.dist [hunk 5]
```diff
@@ -2,18 +2,21 @@
...
 
-    <php>
-        <var name="driver_config_factory" value="Behat\Mink\Tests\Driver\GoutteConfig::getInstance" />
+    <!-- <php>
+        <var name="driver_config_factory" value="Behat\Mink\Driver\Tests\GoutteConfig::getInstance" />
 
-        <!--server name="WEB_FIXTURES_HOST" value="http://test.mink.dev" /-->
-    </php>
+        server name="WEB_FIXTURES_HOST" value="http://test.mink.dev" /
+    </php> -->
```
Comment (aik099) [inline phpunit.xml.dist:19]: Why this change was made?

File: tests/GoutteSimpleTest.php [hunk 6]
```diff
@@ -0,0 +1,86 @@
+<?php
```
Comment (aik099) [inline tests/GoutteSimpleTest.php:1]: Why you've added a new test case? Doesn't the existing test suite work with a new Goutte version?

**Review by aik099 — CHANGES_REQUESTED** [review]

File: composer.json [hunk 7]
```diff
@@ -17,11 +17,13 @@
     "require": {
         "php":                           ">=5.4",
         "behat/mink-browserkit-driver":  "~1.2@dev",
-        "fabpot/goutte":                 "~1.0.4|~2.0|~3.1"
+        "fabpot/goutte":                 "~1.0.4|~2.0|~3.1|^4.0"
```
Comment (aik099) [inline composer.json:20]: Isn't this the only change actually needed to support the new Goutte version?

File: tests/GoutteSimpleTest.php [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+use Behat\Mink\Session;
+use Goutte\Client as GoutteClient;
+use GuzzleHttp\Client as GuzzleHttpClient;
+
+class GoutteSimpleTest extends TestCase
+{
+    public function test_simple_instance()
+    {
+        $driver = new GoutteDriver();
+        $this->assertInstanceOf("Behat\Mink\Driver\GoutteDriver", $driver);
```
Comment (stof) [inline tests/GoutteSimpleTest.php:16]: this is a totally useless test. It is testing PHP

File: tests/GoutteSimpleTest.php [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+        $goutte = new GoutteClient();
+        $goutte->setClient($client);
+        $driver = new GoutteDriver($goutte);
+        $session = new Session($driver);
+        $session->visit("localhost");
+        $status_code = $session->getStatusCode();
+        $this->assertIsInt($status_code);
+    }
+
+    public function test_localhost_visit_html()
```
Comment (stof) [inline tests/GoutteSimpleTest.php:64]: This kind of things is already covered by the driver testsuite (and in a working way, i.e. by visiting a known page rather than hoping that `locahost:80` can be used)

File: composer.json (hunk 3 above)
Comment (stof) [inline composer.json:26]: This guzzle requirement does not make any sense, because Goutte 4 is not based on guzzle at all anymore anyway.

**Review by stof — COMMENTED** [review]
I started the work on BrowserKitDriver 2 to support Symfony 5+ due to the renaming in the BrowserKit component (it will support Symfony 4.4+). Support for Goutte 4 will actually come for free in this new version, as goutte 4 is based on those new versions.

**Comment by stof** [comment]
Closing in favor of #93
